File: boat-sample/src/test/java/sample/xyz/srclab/common/id/IdSample.java
package sample.xyz.srclab.common.id;

import org.testng.annotations.Test;
import xyz.srclab.common.id.StringIdSpec;
import xyz.srclab.common.test.TestLogger;

public class IdSample {

    private static final TestLogger logger = TestLogger.DEFAULT;

    @Test
    public void testId() {
        String spec = "seq-{TimeCount,yyyyMMddHHmmssSSS,1023,%17s%04d}-{Constant,tail}";
        StringIdSpec stringIdSpec = new StringIdSpec(spec);
        //seq-202102071449568890000-tail
        for (int i = 0; i < 10; i++) {
            logger.log(stringIdSpec.newId());
        }
    }
}

File: boat-sample/src/test/kotlin/sample/xyz/srclab/common/id/IdSample.kt
package sample.xyz.srclab.common.id

import org.testng.annotations.Test
import xyz.srclab.common.id.StringIdSpec
import xyz.srclab.common.test.TestLogger

class IdSampleKt {

    @Test
    fun testId() {
        val spec = "seq-{timeCount, yyyyMMddHHmmssSSS, 1023, %17s%04d}-tail";
        val stringIdSpec = StringIdSpec(spec)
        //seq-202102071449568890000-tail
        for (i in 0..9) {
            logger.log(stringIdSpec.create())
        }
    }

    companion object {
        private val logger = TestLogger.DEFAULT
    }
}